File: app/src/main/java/net/zabador/money/service/NotificationServiceAgain.java
package net.zabador.money.service;

import android.os.Bundle;
import android.service.notification.NotificationListenerService;
import android.service.notification.StatusBarNotification;
import android.util.Log;

/**
 * Created by Skye Schneider. The mad king of Domino's
 * in the year of our Lord 8/25/17.
 */

public class NotificationServiceAgain extends NotificationListenerService {

    private static final String TAG = NotificationServiceAgain.class.getSimpleName();

    @Override
    public void onNotificationPosted(StatusBarNotification sbn) {
        Log.d(TAG , "New Notification Posted");

        Bundle extras = sbn.getNotification().extras;

        if(extras != null) {
        }
    }

}

File: app/src/main/java/net/zabador/money/service/NotificationServiceAgain.kt
package net.zabador.money.service

import android.service.notification.NotificationListenerService
import android.service.notification.StatusBarNotification
import android.util.Log

/**
 * Created by Skye Schneider. The mad king of Domino's
 * in the year of our Lord 8/25/17.
 */

class NotificationServiceAgain : NotificationListenerService() {
    
    val MONEY_REGEX = "\\d+\\.\\d{2}"

    override fun onNotificationPosted(sbn: StatusBarNotification) {
        Log.d(TAG, "New Notification Posted")

        val extras = sbn.notification.extras

        if (extras != null) {
        }
    }

    companion object {
        private val TAG = NotificationServiceAgain::class.java.simpleName
    }

}
